### DISCUSSION TITLE
Erro 405 (Método Não Permitido) ao Listar Todas as Categorias

### DISCUSSION THREAD

**Discussion by RPegoli** [balta-io/forum#699] [Tira Dúvidas] [answered]
Olá Balta! Estou fazendo o curso 3074 "Construa um App de Controle Financeiro".
Tenho implementado o código à medida que vou assistindo as aulas e tudo vinha correndo bem até agora, quando me deparei com um problema. Já reassisti várias aulas anteriores para ver se eu deixei passar algo, porém, ainda não encontrei o problema.
No momento estou na aula CRUD - Category - List e quando clico no menu "Categorias" a listagem de categorias não é apresentada e ainda aparece um erro 405 "Método não permitido", conforme imagens em anexo.
A API está respondendo corretamente e por isso creio que o problema está no projeto Dima.WEB.
![API OK](https://github.com/user-attachments/assets/d421c5cd-4e07-4923-9a64-aa80a0f4c884)
![Erro 1](https://github.com/user-attachments/assets/e428d26a-e401-4811-bb60-45e4ea08d96d)

--------------------------------------------------------------------------------------------------------------------------------------------------------
CategoryHandler.cs
using System.Net.Http.Json;
using Dima.Core.Handlers;
using Dima.Core.Models;
using Dima.Core.Requests.Categories;
using Dima.Core.Responses;

namespace Dima.Web.Handlers;

public class CategoryHandler(IHttpClientFactory httpClientFactory) : ICategoryHandler
{
    private readonly HttpClient _client = httpClientFactory.CreateClient(Configuration.HttpClientName);
    public async Task<Response<Category?>> CreateAsync(CreateCategoryRequest request)
    {
        var result = await _client.PostAsJsonAsync("v1/categories", request);
        return await result.Content.ReadFromJsonAsync<Response<Category?>>()
            ?? new Response<Category?>(null, 400, "Falha ao criar a categoria");
    }

    public async Task<Response<Category?>> DeleteAsync(DeleteCategoryRequest request)
    {
        var result = await _client.DeleteAsync($"v1/categories/{request.Id}");
        return await result.Content.ReadFromJsonAsync<Response<Category?>>()
            ?? new Response<Category?>(null, 400, "Falha ao excluir a categoria");
    }

    public async Task<PagedResponse<List<Category>>> GetAllAsync(GetAllCategoriesRequest request)
        => await _client.GetFromJsonAsync<PagedResponse<List<Category>>>("v1/categories")
            ?? new PagedResponse<List<Category>>(null, 400, "Não foi possível obter as categorias");

    public async Task<Response<Category?>> GetByIdAsync(GetCategoryByIdRequest request)
        => await _client.GetFromJsonAsync<Response<Category?>>($"v1/categories/{request.Id}")
            ?? new Response<Category?>(null, 400, "Não foi possível obter a categoria");

    public async Task<Response<Category?>> UpdateAsync(UpdateCategoryRequest request)
    {
        var result = await _client.PutAsJsonAsync($"v1/categories/{request.Id}", request);
        return await result.Content.ReadFromJsonAsync<Response<Category?>>()
            ?? new Response<Category?>(null, 400, "Falha ao atualizar a categoria");
    }
}
--------------------------------------------------------------------------------------------------------------------------------------------------------
List.Razor
@page "/categorias"
@inherits ListCategoriesPage

<MudText Typo="Typo.h3">Categorias</MudText>

<ul>
    @foreach(var cat in Categories)
    {
        <li>@cat.Title</li>
    }
</ul>
--------------------------------------------------------------------------------------------------------------------------------------------------------
List.Razor.cs
using Dima.Core.Handlers;
using Dima.Core.Models;
using Dima.Core.Requests.Categories;
using Microsoft.AspNetCore.Components;
using MudBlazor;

namespace Dima.Web.Pages.Categories;

public partial class ListCategoriesPage: ComponentBase
{
    #region Properties
    public bool IsBusy {get; set;} = false;
    public List<Category>Categories {get; set;} = [];
    #endregion

    #region Services
    [Inject]
    public ISnackbar Snackbar {get; set;} = null!;
    [Inject]
    public ICategoryHandler Handler {get; set;}  = null!;
    #endregion

    #region Overrides
    protected override async Task OnInitializedAsync()
    {
        IsBusy = true;
        try
        {
            var request = new GetAllCategoriesRequest();
            var result = await Handler.GetAllAsync(request);
            if (result.IsSucces)
                Categories = result.Data ?? [];
        }
        catch (Exception ex)
        {
            Snackbar.Add(ex.Message, Severity.Error);
        }
        finally
        {
            IsBusy = false;
        }
    }

    #endregion
}

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @RPegoli como vai?
Este erro indica que está fazendo um request do tipo POST (PUT OU DELETE) para um método GET por exemplo...

Se o seu frontend está OK, pode ser que no seu Backend tenha ficado algum método errado... algum `MapPost` ao invés de `MapGet`.

Você pode (e deve) debugar a aplicação para ter mais detalhes sobre o erro!

**Reply by RPegoli** [reply]
Balta, obrigado pela dica. Encontrei o erro.
Meu MapGet estava esperando receber os parâmetros conforme mostrado na figura.
![image](https://github.com/user-attachments/assets/5024662c-88a0-4cd4-9951-e6fd62cf4260)

**Reply by andrebaltieri** [reply]
Showww 🚀
